canonical/cloud-init
# Ubuntu/bionic

new-upstream-snapshot for release into bionic

To create this branch:
```
git fetch origin   # or upstream your git preference
git checkout origin/ubuntu/bionic -B ubuntu/bionic
uss-tableflip/scripts/new-upstream-snapshot 8377897b

1. manually replace SRU_BUG_NUMBER in the edit view of debian/changelog with 1881018

sed -i -e "1s/UNRELEASED/bionic/" debian/changelog
git commit -m "releasing cloud-init version 20.2-38-g8377897b-0ubuntu1~18.04.1" debian/changelog
```

Because we had a new-upstream-snapshot that we didn't publish, there will be two followup commits here:
1. to re-order the debian/changelog text
```diff
diff --git a/debian/changelog b/debian/changelog
index 68f53d9b..28906560 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,29 +1,5 @@
 cloud-init (20.2-38-g8377897b-0ubuntu1~18.04.1) bionic; urgency=medium
 
-  * refresh patches:
-   + debian/patches/ubuntu-advantage-revert-tip.patch
-  * refresh patches:
-   + debian/patches/openstack-no-network-config.patch
-    - HACKING.rst: update CLA link (#199)
-    - Scaleway: Fix DatasourceScaleway to avoid backtrace (#128)
-      [Louis Bouchard]
-    - cloudinit/cmd/devel/net_convert.py: add missing space (#191)
-    - tools/run-container: drop support for python2 (#192) [Paride Legovini]
-    - Print ssh key fingerprints using sha256 hash (#188)
-    - Make the RPM build use Python 3 (#190) [Paride Legovini]
-    - cc_set_password: increase random pwlength from 9 to 20 (#189)
-    - .travis.yml: use correct Python version for xenial tests (#185)
-    - cloudinit: remove ImportError handling for mock imports (#182)
-    - Do not use fallocate in swap file creation on xfs. (#70) [Eduardo Otubo]
-    - .readthedocs.yaml: install cloud-init when building docs (#181)
-    - Introduce an RTD config file, and pin the Sphinx version to the RTD
-      default (#180)
-    - Drop most of the remaining use of six (#179)
-    - Start removing dependency on six (#178)
-    - Add Rootbox & HyperOne to list of cloud in README (#176) [Adam Dobrawy]
-    - docs: add proposed SRU testing procedure (#167)
-    - util: rename get_architecture to get_dpkg_architecture (#173)
-    - Ensure util.get_architecture() runs only once (#172)
   * d/control: drop pyflakes from Build-Depends
     - This python2 pyflakes package is an unneeded dependency now that tests
       use python3-pyflakes.
@@ -31,6 +7,10 @@ cloud-init (20.2-38-g8377897b-0ubuntu1~18.04.1) bionic; urgency=medium
     - This fixes upstream daily builds.  python3-nose is not removed from
       Build-Depends because, currently, the Ubuntu package builds will still
       use it for testing.
+  * refresh patches:
+   + debian/patches/ubuntu-advantage-revert-tip.patch
+  * refresh patches:
+   + debian/patches/openstack-no-network-config.patch
   * New upstream snapshot. (LP: #1881018)
     - enable Puppet, Chef mcollective in default config (#385)
       [Mina Galić (deprecated: Igor Galić)]
@@ -199,6 +179,26 @@ cloud-init (20.2-38-g8377897b-0ubuntu1~18.04.1) bionic; urgency=medium
     - net/cmdline: correctly handle static ip= config (#201)
       [Dimitri John Ledkov]
     - Replace mock library with unittest.mock (#186)
+    - HACKING.rst: update CLA link (#199)
+    - Scaleway: Fix DatasourceScaleway to avoid backtrace (#128)
+      [Louis Bouchard]
+    - cloudinit/cmd/devel/net_convert.py: add missing space (#191)
+    - tools/run-container: drop support for python2 (#192) [Paride Legovini]
+    - Print ssh key fingerprints using sha256 hash (#188)
+    - Make the RPM build use Python 3 (#190) [Paride Legovini]
+    - cc_set_password: increase random pwlength from 9 to 20 (#189)
+    - .travis.yml: use correct Python version for xenial tests (#185)
+    - cloudinit: remove ImportError handling for mock imports (#182)
+    - Do not use fallocate in swap file creation on xfs. (#70) [Eduardo Otubo]
+    - .readthedocs.yaml: install cloud-init when building docs (#181)
+    - Introduce an RTD config file, and pin the Sphinx version to the RTD
+      default (#180)
+    - Drop most of the remaining use of six (#179)
+    - Start removing dependency on six (#178)
+    - Add Rootbox & HyperOne to list of cloud in README (#176) [Adam Dobrawy]
+    - docs: add proposed SRU testing procedure (#167)
+    - util: rename get_architecture to get_dpkg_architecture (#173)
+    - Ensure util.get_architecture() runs only once (#172)
 
  -- Chad Smith <chad.smith@canonical.com>  Wed, 27 May 2020 17:37:37 -0600
 
```

2. to add a one-time patch/work-item to remove unneeded package deps python3-unittest2 python3-nose
```diff
diff --git a/debian/changelog b/debian/changelog
index 28906560..40a5ecce 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -7,6 +7,8 @@ cloud-init (20.2-38-g8377897b-0ubuntu1~18.04.1) bionic; urgency=medium
     - This fixes upstream daily builds.  python3-nose is not removed from
       Build-Depends because, currently, the Ubuntu package builds will still
       use it for testing.
+  * d/control: drop python3-nose and python3-unittest2 from Build-Depends
+    - Upstream uses pytest now
   * refresh patches:
    + debian/patches/ubuntu-advantage-revert-tip.patch
   * refresh patches:
diff --git a/debian/control b/debian/control
index 92d839bc..ec16a912 100644
--- a/debian/control
+++ b/debian/control
@@ -14,7 +14,6 @@ Build-Depends: debhelper (>= 9.20160709),
                python3-jsonpatch,
                python3-jsonschema,
                python3-mock,
-               python3-nose,
                python3-oauthlib,
                python3-pep8,
                python3-pyflakes,
@@ -23,7 +22,6 @@ Build-Depends: debhelper (>= 9.20160709),
                python3-serial,
                python3-setuptools,
                python3-six,
-               python3-unittest2,
                python3-yaml
 XS-Python-Version: all
 Vcs-Browser: https://git.launchpad.net/cloud-init/?h=ubuntu/devel
```


To test:
```
check diff of your branch vs mine (with out the patches)
# apply the 2 patches
patch -p1 < bionic.patch
patch -p1 < bionic2.patch
git commit -am 'test patches'

run uss-tableflip/scripts/build-package
or make deb to ensure package builds
```
